docs/src/subsystems/node.md

```diff
@@ -32,3 +40,98 @@ Builds all the dependencies in isolation, moving upwards to the top
 package.
 At the end copies over all dependencies into `node_modules` and writes
 symlinks for the bins into `node_modules/.bin`.
+
+### strict (pure + best compatibility) (experimental)
+
+Works almost the same as the granular builder. Not bulletproof stable yet.
+Recommended: Try it out, it should work better than the current default builder, but is not yet released as default.
+
+Features 🌈
+
+- Fully npm compatible
+- No Patches / Overrides required (if "installMethod = copy")
+  - (Most) Complex building of node_modules is fully implemented as python application, because it requires a lot of control flow.
+  - Multiple outputs `["out" "lib" "deps"]` (explained below)
+- Dedicated flattening of node_modules:
+  - Conflicts are resolved during flattening the node_modules folder in favor of the highest semver)
+  - Creates node_modules tree directly from package-lock.json informations. (Through the translator)
+  - Creates node_modules tree with other lock files (such as yarn-lock) most optimal.
+- consume itself:
+  - lets you override/inject a package into your dream2nix project which is built with dream2nix
+
+#### Usage
+
+in `projects.toml` set the `builder` attribute to `'strict-builder'`
+
+> ./projects.toml
+
+```toml
+{{#include ../../../examples/nodejs_alternative_builder/projects.toml}}
+```
+
+#### Override on the `strict-builder`
+
+As currently the builder has three outputs: `$out`, `$lib`, `$deps`.
+When overriding the `installPhase` the user is required to create all three out-paths:
```

Comment:
Sometimes it may be necessary to override the installPhase to build e.g. static page with `webpack` or others like `react`, `vue`, `anuglar` etc. The cleanest way is then to override the installPhase. To create the output he/she wants.
The user is generally not required to have deeper understandings.

Reply:
The example:
```
      packageOverrides = {
        "next-app" = {
          builder = {
            installPhase = ''
              runHook preInstall

              next export
              cp -r out $out

              touch $deps
              touch $lib
              
              runHook postInstall
            '';
          };
        };
      };
    };
}
```

builds static page / works for `create next app`

```
johannes@johannes-nixos ~/g/d/next-app (main)> tree result/ -a -L 2
result/
├── 404.html
├── favicon.ico
├── index.html
├── _next
│   ├── 61CZE-ePDegW-aHRoz1tY
│   └── static
├── next.svg
├── nix-support
│   └── propagated-build-inputs
├── thirteen.svg
└── vercel.svg
```


Without that override the installPhase would create other folders and stuff, which is unwanted. (If you have custom builds)

Reply:
The user should not need to add build logic.
How about:
```
postBuild = "next export";
```
?

Reply:
*the user should not need to copy paste code in order to add build logic